### PR TITLE
Update build to .NET 6

### PR DESCRIPTION
**Pull request by tillig**
Follows the same updates as the previous project updates. No functionality changes, just build and `.editorconfig` changes.

- Build using .NET 6, still targeting the original frameworks.
- Updates to analyzers plus analyzer fixes.
- Updates to dependencies _except_ the `ServiceFabric` dependencies.
- File-scoped namespaces.
- Added missing documentation.
- Updated README format, adding it to the NuGet package.
- Packaging symbols as .snupkg.
- Using VS Code compatible .resx generation.

### CODE REVIEW

**Comment by tillig** [comment]
@alistairjevans Let me know if you don't have this kind of time. I'm fine merging it back to `develop` without a deep review; we've done this a few times now.

**Comment by alistairjevans** [comment]
I should be able to take a look tomorrow.

File: Directory.Build.targets [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (alistairjevans) [inline Directory.Build.targets]: Do you need a `Directory.Build.targets` if it's empty?

File: README.md [hunk 2]
```diff
@@ -2,10 +2,77 @@
...
 Service Fabric support for [Autofac](https://autofac.org).
 
-[![Build status](https://ci.appveyor.com/api/projects/status/pwrw1chyf0c2hlj1?svg=true)](https://ci.appveyor.com/project/Autofac/autofac-servicefabric) [![Open in Visual Studio Code](https://open.vscode.dev/badges/open-in-vscode.svg)](https://open.vscode.dev/autofac/Autofac.ServiceFabric)
+[![Build status](https://ci.appveyor.com/api/projects/status/pwrw1chyf0c2hlj1?svg=true)](https://ci.appveyor.com/project/Autofac/autofac-servicefabric)
 
 Please file issues and pull requests for this package in this repository rather than in the Autofac core repo.
 
 - [Documentation](https://autofac.readthedocs.io/en/latest/integration/servicefabric.html)
-- [NuGet](https://www.nuget.org/packages/Autofac.ServiceFabric)
+- [NuGet](https://www.nuget.org/packages/Autofac.WebApi2/)
```
Comment (alistairjevans) [inline README.md]: Incorrect repo reference.
```suggestion
- [NuGet](https://www.nuget.org/packages/Autofac.ServiceFabric/)
```

**Review by alistairjevans — CHANGES_REQUESTED** [review]
Just the one copy-paste issue (and a question about an empty targets file).

File: Directory.Build.targets (hunk 1 above)
Comment (tillig) [reply Directory.Build.targets]: That's a good question. We had it in core Autofac so I just brought it over here, but... let me test. I could be just cargo-cult-copy/pasting here.

Comment (tillig) [reply Directory.Build.targets]: Turns out it's not necessary. I'll nuke it.

**Review by alistairjevans — APPROVED** [review]
🚢

**Pull request merged**